pkg/registry/controller/rest.go

```diff
@@ -47,14 +48,25 @@ func (rcStrategy) NamespaceScoped() bool {
 func (rcStrategy) PrepareForCreate(obj runtime.Object) {
 	controller := obj.(*api.ReplicationController)
 	controller.Status = api.ReplicationControllerStatus{}
+
+	// We could allow users to initialize the sequenceNumbers, but since
+	// we don't allow them to update it, doing so would be non-intuitive.
+	controller.Spec.SequenceNumber = 0
+	controller.Status.SequenceNumber = 0
 }
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (rcStrategy) PrepareForUpdate(obj, old runtime.Object) {
-	// TODO: once RC has a status sub-resource we can enable this.
-	//newController := obj.(*api.ReplicationController)
-	//oldController := old.(*api.ReplicationController)
-	//newController.Status = oldController.Status
+	newController := obj.(*api.ReplicationController)
+	oldController := old.(*api.ReplicationController)
+
+	// Any changes to the spec increment the sequence number, any changes to the
+	// status should reflect the sequence number of the corresponding spec. We push
+	// the burden of managing the status onto the clients to avoid confusion when
+	// we have concurrent updates.
+	if !reflect.DeepEqual(oldController.Spec, newController.Spec) {
```

Comment:
In principle, at least. I suppose for now deletionTimestamp doesn't do anything for replication controllers, and labels and annotations have no effect, either, so just checking spec is ok.